cmd/agent/subcommands/streamlogs/command.go

```diff
@@ -166,3 +172,18 @@ func openFileForWriting(filePath string) (*os.File, *bufio.Writer, error) {
 	bufWriter := bufio.NewWriter(f) // default 4096 bytes buffer
 	return f, bufWriter, nil
 }
+
+// StreamLogs is a public function that can be used by other packages to stream logs.
+func StreamLogs(log log.Component, config config.Component, cliParams *CliParams) error {
+	dir := filepath.Dir(cliParams.FilePath)
+
+	// Create the directory if it does not exist
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
```

Comment:
Does it make sense to put the directory management code in `streamLogs` instead of duplicating it here and in `PreRunE`?